library/Icingadb/Common/TicketLinks.php

```diff
@@ -5,11 +5,41 @@
 namespace Icinga\Module\Icingadb\Common;
 
 use Icinga\Application\Hook;
+use Icinga\Application\Hook\TicketHook;
 
 trait TicketLinks
 {
     /** @var bool */
-    protected $ticketLinkEnabled = false;
+    protected $ticketLinkEnabled = false; // TODO: Remove once all usages are removed
+
+    /** @var bool Whether the list item is being rendered in the detailed view */
+    protected $isDetailView = false;
+
+    /**
+     * Set whether the list item is being rendered in the detailed view
+     *
+     * If set to true, the ticket link will be created
+     *
+     * @param bool $state
+     *
+     * @return $this
+     */
+    public function setIsDetailView(bool $state = true): self
+    {
+        $this->isDetailView = $state;
+
+        return $this;
+    }
+
+    /**
+     * Get whether the list item is being rendered in the detailed view
+     *
+     * @return bool
+     */
+    public function isDetailView(): bool
+    {
+       return $this->isDetailView;
+    }
```

Comment:
Drop this. A detail uses the trait method `createTicketLinks` in which case ticket links are **always** expected.
An item renderer need to be told in order to do so, but that's a feature of the renderer, not the trait.